### DISCUSSION TITLE
Can we standardize logical direction of rule options?

### DISCUSSION THREAD

**Discussion by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6101] [Technical Discussions]
Some rule options start with words like `allow` or `ignore` and make the rule more lenient. For example:

* 👐 [`@typescript-eslint/prefer-readonly-parameter-types`](https://typescript-eslint.io/rules/prefer-readonly-parameter-types) > [`ignoreInferredTypes `](https://typescript-eslint.io/rules/prefer-readonly-parameter-types/#ignoreinferredtypes)
* 👐 [`@typescript-eslint/restrict-plus-operands`](https://typescript-eslint.io/rules/restrict-plus-operands) > [`allowAny`](https://typescript-eslint.io/rules/restrict-plus-operands/#allowany)

Other rule options start with words like `check` and make the rule more strict. For example:

* 🙅 [`@typescript-eslint/prefer-readonly-parameter-types`](https://typescript-eslint.io/rules/prefer-readonly-parameter-types) > [`checkParameterProperties`](https://typescript-eslint.io/rules/prefer-readonly-parameter-types/#checkparameterproperties)
* 🙅 [`@typescript-eslint/restrict-plus-operands`](https://typescript-eslint.io/rules/restrict-plus-operands) > [`checkCompoundAssignments`](https://typescript-eslint.io/rules/restrict-plus-operands/#checkcompoundassignments)

Some rules include _both_ options styles - including those two! 😩 This makes it hard to keep track of them.

Can I propose we prefer the 👐 **`allow`**-prefixed names, and have their default values be `false`? That way it won't be a breaking change to add them.

**Comment by bradzacher** [comment]
In those cases "allow" wouldn't make sense because the rule isn't allowing something - it's expanding its set of checks.

You need two verbs to describe options I think:
- "check" - turn on additional features in the rule.
- "allow" OR "ignore" (we could pick one) - completely turn off a class of things from a check without disabling the check.
